src/database/transaction.rs

```diff
@@ -213,32 +213,32 @@ impl DatabaseTransaction {
             InnerConnection::MySql(ref mut c) => {
                 <sqlx::MySql as sqlx::Database>::TransactionManager::rollback(c)
                     .await
-                    .map_err(sqlx_error_to_query_err)?
+                    .map_err(sqlx_error_to_query_err)
             }
             #[cfg(feature = "sqlx-postgres")]
             InnerConnection::Postgres(ref mut c) => {
                 <sqlx::Postgres as sqlx::Database>::TransactionManager::rollback(c)
                     .await
-                    .map_err(sqlx_error_to_query_err)?
+                    .map_err(sqlx_error_to_query_err)
             }
             #[cfg(feature = "sqlx-sqlite")]
             InnerConnection::Sqlite(ref mut c) => {
                 <sqlx::Sqlite as sqlx::Database>::TransactionManager::rollback(c)
                     .await
-                    .map_err(sqlx_error_to_query_err)?
+                    .map_err(sqlx_error_to_query_err)
             }
             #[cfg(feature = "mock")]
-            InnerConnection::Mock(ref mut c) => {
-                c.rollback();
-            }
-        }
+            InnerConnection::Mock(ref mut c) => c.rollback(),
+            #[allow(unreachable_patterns)]
+            _ => Err(conn_err("Disconnected")),
+        }?;
         self.open = false;
         Ok(())
     }
 
     // the rollback is queued and will be performed on next async operation, like returning the connection to the pool
     #[instrument(level = "trace")]
-    fn start_rollback(&mut self) {
+    fn start_rollback(&mut self) -> Result<(), DbErr> {
```

Comment:
This also should not be changed

Reply:
This is private method of a struct impl. So it should be fine.